bob/src/cluster/operations.rs

```diff
@@ -26,33 +48,34 @@ fn is_result_successful<TErr: Debug>(
     0
 }
 
-pub(crate) async fn finish_at_least_handles<TErr: Debug>(
-    handles: &mut Tasks<TErr>,
+pub(crate) async fn finish_at_least_handles<TRes: AffectedReplicasProvider, TErr: Debug>(
+    handles: &mut Tasks<TRes, TErr>,
     at_least: usize,
-) -> Vec<NodeOutput<TErr>> {
+) -> (Vec<NodeOutput<TRes>>, Vec<NodeOutput<TErr>>) {
     let mut ok_count = 0;
     let mut errors = Vec::new();
+    let mut oks = Vec::new();
```

Comment:
I suggest to create this vector with initial capacity = `at_least.min(handles.len())`